### PR TITLE
Limit each duplicate declaration error to 32 references

### LINKED ISSUE #2890: Overload conflict warnings (constructors and functions) should limit the number of secondary declarations
If the conflicting number of declarations is over a certain threshold (lets say 32), truncate the list and include a new error explaining the list is truncated, because there more than 32 conflicts and also display the actual number of conflicts (since it is available).

Part of #2876.

### PR DESCRIPTION
**Pull request by axic**
Fixes #2890. Depends on #2888.

### CODE REVIEW

File: libsolidity/analysis/TypeChecker.cpp [hunk 1]
```diff
@@ -164,28 +164,52 @@ void TypeChecker::checkContractDuplicateFunctions(ContractDefinition const& _con
 		for (; it != functions[_contract.name()].end(); ++it)
 			ssl.append("Another declaration is here:", (*it)->location());
 
+		string msg = "More than one constructor defined.";
+		size_t occurances = ssl.infos.size();
+		if (occurances > 32)
+		{
+			ssl.infos.resize(32);
+			msg += " The list is truncated from " + boost::lexical_cast<string>(occurances) + " to 32 occurances.";
```
Comment (axic) [inline libsolidity/analysis/TypeChecker.cpp]: Change to `occurrences`.

File: libsolidity/analysis/TypeChecker.cpp [hunk 2]
```diff
@@ -164,28 +164,52 @@ void TypeChecker::checkContractDuplicateFunctions(ContractDefinition const& _con
...
+				}
+
+			if (ssl.infos.size() > 0)
+			{
+				string msg = "Function with same name and arguments defined twice.";
+				size_t occurances = ssl.infos.size();
+				if (occurances > 32)
+				{
+					ssl.infos.resize(32);
+					msg += " The list is truncated from " + boost::lexical_cast<string>(occurances) + " to 32 occurances.";
```
Comment (axic) [inline libsolidity/analysis/TypeChecker.cpp]: Change to `occurrences`.

File: libsolidity/analysis/TypeChecker.cpp [hunk 3]
```diff
@@ -164,10 +164,18 @@ void TypeChecker::checkContractDuplicateFunctions(ContractDefinition const& _con
 		for (; it != functions[_contract.name()].end(); ++it)
 			ssl.append("Another declaration is here:", (*it)->location());
 
+		string msg = "More than one constructor defined.";
+		size_t occurances = ssl.infos.size();
+		if (occurances > 32)
+		{
+			ssl.infos.resize(32);
+			msg += " The list is truncated from " + boost::lexical_cast<string>(occurances) + " to 32 occurances.";
```
Comment (chriseth) [inline libsolidity/analysis/TypeChecker.cpp]: Perhaps shorter: `(only listing the first 32 occurrences)`?

**Review by chriseth — APPROVED** [review]

File: libsolidity/analysis/TypeChecker.cpp (hunk 3 above)
Comment (axic) [reply libsolidity/analysis/TypeChecker.cpp]: Shorted it to `Truncated from X to the first 32 occurrences`

Alternative: `Only showing the first 32 occurrences out of a total X.`

I think it is useful to show the total.

**Pull request merged**